packages/stencil-library/src/components/button/button.e2e.ts

```diff
@@ -1,105 +1,187 @@
 import { E2EElement, newE2EPage } from '@stencil/core/testing';
+import { Mode, ButtonVariant, IconPosition } from '../../interface';
 
 const renderAndGetButtonElement = async (content: string): Promise<E2EElement> => {
-  const page = await newE2EPage();
-  await page.setContent(content);
-  const button = await page.find('njwds-button > button');
-  return button;
+    const page = await newE2EPage();
+    await page.setContent(content);
+    const button = await page.find('njwds-button > button');
+    return button;
 };
 
+const getElementClasses = (element: E2EElement): string[] => {
+    return element.className.split(' ').sort();
+};
+
+const variant: Record<ButtonVariant, ButtonVariant> = {
```

Comment:
[dust] I really like the pattern of using Record objects for variant, etc. to make the tests clearer!